R/multiple-conditions.R

```diff
@@ -0,0 +1,182 @@
+#' @title Multiple chronic conditions
+#' 
+#' @description This function generates a derived variable (multiple_conditions)
+#'  that counts the number of chronic conditions a respondent has. This function
+#'  takes 5 CCHS variables (heart disease, cancer, stroke, bowel disorder, and
+#'  arthritis), and well one derived variable (respiratory condition) to count
+#'  the number of conditions a respondent has.
+#'  
+#' @param CCC_121 variable indicating if respondent has heart disease
+#' 
+#' @param CCC_131 variable indicating if respondent has active cancer
+#' 
+#' @param CCC_151 variable indicating if respondent suffers from the effects
+#'  of a stroke
+#' 
+#' @param CCC_171 variable indicating if respondent has a bowel disorder
+#' 
+#' @param resp_condition_der derived variable indicating if respondent has a
+#'  respiratory condition. See \code{\link{resp_condition_fun1}} for
+#'  documentation on how variable was derived
+#'
+#' @param CCC_051 variable indicating if respondent has arthritis or
+#'  rheumatism
+#'
+#' @details mood disorder (CCC_280) was not asked to respondents in the 2001
+#'  CCHS survey cycle. This mean respondents in this cycle will only be able to
+#'  have a maximum of 6 chronic conditions as opposed to 7 for respondents in
+#'  other cycles. \code{\link{multiple_conditions_fun2}} is used for CCHS cycles
+#'  from 2003 to 2014.
+#' 
+#' @return A categorical variable indicating the number of chronic conditions
+#'  a respondent has. Respondents with 5 or more conditions are grouped in the
+#'  "5+" category. 
+#' 
+#' @examples 
+#'  # Using rec_with_table() to generate multiple_conditions in a CCHS
+#'  # cycle.
+#'  
+#'  # multiple_conditions_fun1() is specified in variable_details.csv along with
+#'  # the CCHS variables and cycles included.
+#'  
+#'  # To generate multiple_conditions, use rec_with_table() and specify the
+#'  # multiple_conditions, along with the variables that are derived from it.
+#'  # Since resp_condition_der is also a derived variable, you will have to
+#'  # specify the variables that are derived from it. In this example, data
+#'  # from the 2001 CCHS will be used, so DHHGAGE_cont, CCC_091, and CCC_91A,
+#'  # and CCC_031 will be specified along with resp_condition_der.
+#'  
+#' library(cchsflow)
+#'  conditions_2001 <- rec_with_table(cchs2001_p, c("DHHGAGE_cont", "CCC_091",
+#'  "CCC_91A", "CCC_031", "CCC_121","CCC_131","CCC_151", "CCC_171","CCC_280",
+#'  "resp_condition_der","CCC_051", "multiple_conditions"))
+#'  
+#'  head(conditions_2001)
+#' 
+#' @seealso \code{\link{multiple_conditions_fun2}}     
+#' @export
+
+multiple_conditions_fun1 <- 
+  function(CCC_121,CCC_131,CCC_151,CCC_171,resp_condition_der,CCC_051){
```

Comment:
spacing. Make conditions default attributions? CCC_121 = CCC_121 etc.

Reply:
I'm hesitant to add default attributions. The reason being is that while the main use of the two functions will be to generate the `multiple_conditions` variable via `rec_with_table()`, in the event that a user wants to generate the variable with inputted values, we would want them to input all parameter to minimize error. I could be wrong, however.

Reply:
I'm not quite following, and it may be because I am off mark.
Yes, a user could use the variable with new imputted valudes. I that case they just replace the defaults with the new variable or imputted value?

Reply:
I dont see how would we even obtain the default values
This function is used withing rec_with_table and passed individual values and has no access to the data

Reply:
OK. You convinced me. We also have all the conditions in variable_details and so we are transparent and don't require users to input those arguments. So, people should be ok for our main uses. 

DerivedVar::[CCC_121, CCC_131, CCC_151, CCC_171, resp_condition_der, CCC_051]